### ISSUE TITLE
Sender timeout has no effect

### ISSUE THREAD

**Issue by yunhaoling** [Azure/azure-sdk-for-python#5307]
@mbrancato commented on [Mon Apr 08 2019](https://github.com/Azure/azure-event-hubs-python/issues/105)

When setting `send_timeout` using `EventHubClient.add_sender()`, the timeout doesn't seem to be enforced when calling `Sender.send()`. This may be upstream issue with uamqp but i'm not sure if there is a better goal of implementing it here, not upstream.

---

@annatisch commented on [Mon Apr 08 2019](https://github.com/Azure/azure-event-hubs-python/issues/105#issuecomment-480881647)

Thanks for the report @mbrancato!
Could I please see a snippet of how you're using the setting?

---

@mbrancato commented on [Mon Apr 08 2019](https://github.com/Azure/azure-event-hubs-python/issues/105#issuecomment-480917114)

Sure, below is a minimal program that does not timeout properly if the network connection is interrupted.

First, the pip packages I'm using that are probably relevant:
```
azure-common             1.1.18
azure-eventhub           1.3.1
azure-nspkg              3.0.2
azure-storage            0.36.0
azure-storage-blob       1.5.0
azure-storage-common     1.4.0
uamqp                    1.1.0
```

minimal example:
```python
import time
import random
import json
import logging
from azure.eventhub import EventHubClient, EventData

eventHubNamespace = "..."
eventHubName = "..."
eventHubPolicy = "..."
eventHubKey = "..."
eventHubAddress = "amqps://%s.servicebus.windows.net/%s" % (eventHubNamespace, eventHubName)

logging.basicConfig(format="%(asctime)-15s - %(message)s")
logging.warning("Creating client")
outputClient = EventHubClient(eventHubAddress, debug=False, username=eventHubPolicy, password=eventHubKey)
logging.warning("Adding sender")
outputSender = outputClient.add_sender(partition=None, send_timeout=1)
logging.warning("Running client")
outputClient.run()

# A network connection issue, etc after here will not trigger the timeout

while True:
    events = []
    try:
        for i in range(random.randint(1,5)):
            event = {}
            event["value"] = i
            events.append(event)

        logging.warning("Generated %d events" % (len(events)))

        logging.warning("Shipping events")
        outputSender.send(EventData(json.dumps(events)))

        logging.warning("Sleeping")
        time.sleep(4)
    except KeyboardInterrupt:
        logging.warning("Stopping client")
        outputClient.stop()
        exit(0)
```

**Comment by gizmo93** [comment]
Hi there,
is there already a fix for this issue? We notice some strange and not really reproducible behavior in our application, where the sender sometimes just seems to wait forever for a message to be sent because of some network related errors and the sender timeout does not seem to work here. We work around this issue by killing and restarting the process automatically after a period without reaction, but I think it would be great if this timeout setting would work.

**Comment by yunhaoling** [comment]
Hello @gizmo93 ,

Can you try our latest preview eventhubs sdk to see if the issue still happens?
https://pypi.org/project/azure-eventhub/5.0.0b3/

**Comment by yunhaoling** [comment]
@gizmo93 , we've released azure-eventhub 1.3.3 along with uamqp v1.2.5 which should fix the send timeout problem. Please try the using the latest library.

**Comment by yunhaoling** [comment]
@gizmo93 ,I'm closing this issue, but feel free to open the issue if problem still exists.
